File: library/src/main/java/me/proxer/library/api/info/EntryCoreEndpoint.java
package me.proxer.library.api.info;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;
import me.proxer.library.entity.info.Entry;
import me.proxer.library.entity.info.EntryCore;

/**
 * Endpoint for retrieving the most important information of an {@link Entry}.
 * <p>
 * If you need everything, consider using the {@link EntryEndpoint}.
 *
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public final class EntryCoreEndpoint implements Endpoint<EntryCore> {

    private final InternalApi internalApi;

    private final String id;

    EntryCoreEndpoint(final InternalApi internalApi, final String id) {
        this.internalApi = internalApi;
        this.id = id;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<EntryCore> build() {
        return internalApi.entryCore(id);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/info/EntryCoreEndpoint.kt
package me.proxer.library.api.info

import me.proxer.library.api.Endpoint
import me.proxer.library.api.ProxerCall
import me.proxer.library.entity.info.Entry
import me.proxer.library.entity.info.EntryCore

/**
 * Endpoint for retrieving the most important information of an [Entry].
 *
 * If you need everything, consider using the [EntryEndpoint].
 *
 * @author Ruben Gees
 */
class EntryCoreEndpoint internal constructor(
    private val internalApi: InternalApi,
    private val id: String
) : Endpoint<EntryCore> {

    override fun build(): ProxerCall<EntryCore> {
        return internalApi.entryCore(id)
    }
}
